src/main/java/com/google/idea/perf/methodtracer/Argument.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.idea.perf.methodtracer
+
+import java.util.*
+
+class Argument(val value: Any?, val index: Byte) {
```

Comment:
Since instrumented bytecode will instantiate `Argument` directly, the `Argument` class should exist in the agent (on the boot class path).

You can see why by trying this command:
```
trace java.util.concurrent.ConcurrentHashMap#get[1]
```
You will get `NoClassDefFoundError` because the bytecode in `ConcurrentHashMap` tries to reference the `Argument` class (which it does not have access to).

If we put the `Argument` class back into the agent, then we can fix the tests by either (1) attaching the agent before the test runs, or (2) editing`build.gradle.kts` to load the agent at startup via a JVM flag.

Reply:
👍